source/server/listener_manager_impl.cc

```diff
@@ -261,25 +271,38 @@ bool ListenerManagerImpl::addOrUpdateListener(const Json::Object& json) {
 }
 
 void ListenerManagerImpl::drainListener(ListenerImplPtr&& listener) {
-  // TODO(mattklein123): Actually implement timed draining in a follow up. Currently we just
-  // correctly synchronize removal across all workers.
+  // First add the listener to the draining list. Thist must be done under the lock since it
```

Comment:
It no longer applies. The drain manager handles the timer and then removal via the completion callback. I will add more comments.